src/common/recordbatch/src/filter.rs

```diff
@@ -144,13 +151,42 @@ impl SimpleFilterEvaluator {
     }
 }
 
+/// copy from datafusion::physical_plan::src::filter.rs
+pub fn batch_filter(
+    batch: &RecordBatch,
+    predicate: &Arc<dyn PhysicalExpr>,
+) -> DfResult<RecordBatch> {
+    predicate
+        .evaluate(batch)
+        .and_then(|v| v.into_array(batch.num_rows()))
+        .and_then(|array| {
+            let filter_array = match as_boolean_array(&array) {
+                Ok(boolean_array) => Ok(boolean_array.to_owned()),
```

Comment:
Prefer to use `clone` to indicate the operation explicitly.
```suggestion
                Ok(boolean_array) => Ok(boolean_array.clone()),
```